100=OptionsHandler.callHandlerMethod()
101=An unexpected exception: {} has occurred, producing a series of exceptions: {}

200=OptionsManager.__init__()
201=OptionManger has been correctly initialized.

210=OptionsManager.loadSetting()
211=OptionManger has been correctly reloaded.

220=OptionsManager.saveSetting()
221=OptionManger saved options {} and logging {}.


1000=Connect URL ERROR
1001=Authentication ERROR
1002=Authorization ERROR
1003=Setup ERROR
1004=Network ERROR
1005=DataBase ERROR
1006=Provider ERROR


1100=Driver.__init__()
1101=Driver initialization completed

1110=Driver.connect()
1111=Loading the driver for the URL {}
1112=Invalid protocol <{}>
1113=Invalid user account and/or password.
1114=Invalid user account <{}>
1115=Hsqldb version {} database is loaded, the user {} is connected.
1116=An error <{}> occurred while connecting with trace <{}>.

1120=Driver._getDataSource()
1121=The extension {} was not found. You must install this extension to use {}.
1122=Version {} of extension {} is obsolete. You need to update this extension to version: {}.
1123=Unable to connect to URL {}. An SQLException was thrown with the message: {}.
1124=HsqlDB driver version {} is obsolete. You need to update this driver to version {} or higher.


1200=DataSource.__init__()
1201=The data source is loading...
1202=Data source successfully loaded.

1250=DataBase.__init__()
1251=Database does not exist, creating new database.
1252=Database creation was successful.

1300=Provider.insertUser()
1301=User <{}> does not exist, requesting ID from provider.
1302=The provider provided <{}> as an identifier for user <{}>.

1320=Provider.initAddressbooks()
1321=Initializing user <{}>.
1322=Initialization for user <{}> was successful.

1330=Provider.initUserBooks()
1331=Initializing address book for user <{}>.
1332=Address book initialization for user <{}> was successful.

1340=Provider.initUserGroups()
1341=Initializing address group for user <{}>.
1342=Address group initialization for user <{}> was successful.

1350=Provider.initUserGroup()
1351=Initializing address group for book <{}>.
1352=Address group initialization for book <{}> was successful.


1350=User.__init__()
1351=User <{}> is loading...
1352=Searching for user <{}> in the database.
1353=User <{}> exists in the database. Trying to connect to the provider.
1354=User <{}> does not exist in the database. Trying to connect to the provider.
1355=User <{}> successfully loaded.


1400=DataSource.getConnection()
1401=The user {} has abandoned the OAuth2 authorization wizard, the {} extension cannot continue without this authorization!!!


1500=User.__init__()
1501=The user {} has abandoned the OAuth2 authorization wizard, the {} extension cannot continue without this authorization!!!
1502=Can't connect to server <{}>. Is the network offline?
1503=Can't insert user {} in DataBase


1600=Provider.raiseForStatus()
1601=HTTP request <{}> returned status code {} for user {} on URL {} with content {}

1610=Provider.initUserBooks()
1611=Can't retrieve AddressBook for user {}, the server {} don't support AddressBook!!!

1620=Provider._getDiscoveryUrl()
1621=HTTP request <{}> for user {} on URL {} cannot retrieve <Location> on headers {}
1622=Despite {} attempts, impossible to find a discovery url other than: {}

1640=Provider._getNewUserId()
1641=User {} with password {} cannot connect to server {} at URL {}
1642=User {} cannot find on URL {} a CardDAV compatible server

1650=Provider._supportAddressbook()
1651=HTTP request <{}> for user {} on URL {} cannot retrieve <DAV> on headers {}

1660=Provider._getAddressbooksUrl()
1661=HTTP request <{}> for user {} on URL {} cannot parse content {}

